File: android/java/StorylyDemo/app/src/main/java/com/appsamurai/storylydemo/IntegrationActivity.java
package com.appsamurai.storylydemo;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.appsamurai.storyly.Story;
import com.appsamurai.storyly.StoryComponent;
import com.appsamurai.storyly.StoryGroup;
import com.appsamurai.storyly.StorylyDataSource;
import com.appsamurai.storyly.StorylyInit;
import com.appsamurai.storyly.StorylyListener;
import com.appsamurai.storyly.StorylyView;
import com.appsamurai.storyly.analytics.StorylyEvent;

import java.util.List;

public class IntegrationActivity extends AppCompatActivity {
    private static final String STORYLY_DEMO_TOKEN = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjU1NiwiYXBwX2lkIjoxMzg5LCJpbnNfaWQiOjEwNDA5fQ.kXqBdpUcKaJe7eA98PqHahMDf-123Uhb82t_mYzbBUM";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_integration);

        StorylyView storylyView = findViewById(R.id.storyly_view);
        storylyView.setStorylyInit(new StorylyInit(STORYLY_DEMO_TOKEN));
        storylyView.setStorylyListener(new StorylyListener() {
            @Override
            public void storylyActionClicked(@NonNull StorylyView storylyView, @NonNull Story story) {
                Log.d("[storyly]", "IntegrationActivity:storylyActionClicked - story {"+story+"}");
                try {
                    String actionUrl = story.getMedia().getActionUrl();
                    if (actionUrl == null) return;

                    Log.d("[storyly]", "IntegrationActivity:storylyActionClicked - forwarding to url {"+actionUrl+"}");
                    Intent intent = new Intent(Intent.ACTION_VIEW);
                    intent.setData(Uri.parse(actionUrl));
                    startActivity(intent);
                } catch (Exception exception) {
                    Toast.makeText(getBaseContext(), exception.getLocalizedMessage(), Toast.LENGTH_LONG).show();
                }
            }

            @Override
            public void storylyLoaded(@NonNull StorylyView storylyView, @NonNull List<StoryGroup> storyGroupList, @NonNull StorylyDataSource storylyDataSource) {
                Log.d("[storyly]", "IntegrationActivity:storylyLoaded - storyGroupList size {"+storyGroupList.size()+"} - source {"+storylyDataSource+"}");
            }

            @Override
            public void storylyLoadFailed(@NonNull StorylyView storylyView, @NonNull String errorMessage) {
                Log.d("[storyly]", "IntegrationActivity:storylyLoadFailed - error {"+errorMessage+"}");
            }

            @Override
            public void storylyStoryShown(@NonNull StorylyView storylyView) {}

            @Override
            public void storylyStoryShowFailed(@NonNull StorylyView storylyView, @NonNull String errorMessage) {}

            @Override
            public void storylyStoryDismissed(@NonNull StorylyView storylyView) {}

            @Override
            public void storylyUserInteracted(@NonNull StorylyView storylyView, @NonNull StoryGroup storyGroup, @NonNull Story story, @NonNull StoryComponent storyComponent) {}

            @Override
            public void storylyEvent(@NonNull StorylyView storylyView, @NonNull StorylyEvent storylyEvent, @Nullable StoryGroup storyGroup, @Nullable Story story, @Nullable StoryComponent storyComponent) {}
        });
    }
}

File: android/kotlin/StorylyDemo/app/src/main/java/com/appsamurai/storylydemo/IntegrationActivity.kt
package com.appsamurai.storylydemo

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.appsamurai.storyly.*
import com.appsamurai.storylydemo.databinding.ActivityIntegrationBinding


class IntegrationActivity : AppCompatActivity() {
    companion object {
        const val STORYLY_DEMO_TOKEN = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjU1NiwiYXBwX2lkIjoxMzg5LCJpbnNfaWQiOjEwNDA5fQ.kXqBdpUcKaJe7eA98PqHahMDf-123Uhb82t_mYzbBUM"
    }

    private lateinit var binding: ActivityIntegrationBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityIntegrationBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.storylyView.storylyInit = StorylyInit(STORYLY_DEMO_TOKEN)
        binding.storylyView.storylyListener = object : StorylyListener {
            override fun storylyLoaded(
                storylyView: StorylyView,
                storyGroupList: List<StoryGroup>,
                dataSource: StorylyDataSource
            ) {
                Log.d("[storyly]", "IntegrationActivity:storylyLoaded - storyGroupList size {${storyGroupList.size}} - source {$dataSource}")
            }

            override fun storylyLoadFailed(
                storylyView: StorylyView,
                errorMessage: String
            ) {
                Log.d("[storyly]", "IntegrationActivity:storylyLoadFailed - error {$errorMessage}")
            }

            override fun storylyActionClicked(
                storylyView: StorylyView,
                story: Story
            ) {
                Log.d("[storyly]", "IntegrationActivity:storylyActionClicked - story {$story}")
                try {
                    story.media.actionUrl?.let { url ->
                        Log.d("[storyly]", "IntegrationActivity:storylyActionClicked - forwarding to url {$url}")
                        startActivity(
                            Intent(Intent.ACTION_VIEW).apply {
                                data = Uri.parse(url)
                            }
                        )
                    }
                } catch (exception: Exception) {
                    Toast.makeText(baseContext, exception.localizedMessage, Toast.LENGTH_LONG).show()
                }
            }
        }
    }
}
